cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -118,6 +124,20 @@ func (m *Manager) Deploy(ctx context.Context) (*DeployResult, error) {
 		}
 	}
 
+	sqlServerOperations, err := azdSqlServerOperations(model, filepath.Join(m.projectPath, m.options.Path))
+	if !azdOperationsEnabled && len(sqlServerOperations) > 0 {
+		m.console.Message(ctx, ErrSqlScriptOperationDisabled.Error())
+	}
+	if azdOperationsEnabled {
+		if err != nil {
+			return nil, fmt.Errorf("looking for azd sql scripts operations: %w", err)
+		}
+		if err := doSqlScriptOperation(
+			ctx, sqlServerOperations, m.console, *m.env, m.sqlCmdCli); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
We should not start injecting special case scenarios like sqlcmd related login into purposely generic components like the provisioning manager.  We need to come up with a scalable design that can provide clear single responsibility patterns in our components.